src/open-id-connect-navigation-strategies.ts

```diff
@@ -36,11 +35,7 @@ export default class OpenIdConnectNavigationStrategies {
     };
 
     const navigationInstruction = () => {
-      // Use location.assign not instruction.config.redirect,
-      // because the former adds the route to the web browser's history,
-      // and that controls what will load on a page refresh.
-      // See https://github.com/aurelia-contrib/aurelia-open-id-connect/issues/46
-      this.$window.location.assign(redirectRoute);
+      instruction.config.redirect = redirectRoute;
```

Comment:
This does resolve the error message but then breaks the page refresh. I wonder how we can resolve both problems at the same time.

Reply:
What exactly do you mean by breaking the page refresh? Is it #46? I was unable to reproduce that bug from the outlined steps.

Reply:
We might be able to do something like this first: 
```
window.location.pushState(undefined, undefined, redirectRoute);
```

Reply:
or maybe something like `history.navigate(redirectRoute, {trigger: false, replace: true})`, see (https://github.com/aurelia/history-browser/blob/master/src/index.js#L126)

Reply:
Can you reproduce #46?

Reply:
I just tested and an approach like this seems to work: 
```
this.$window.history.pushState(undefined, 'TODO', redirectRoute);
instruction.config.redirect = redirectRoute;                     
```

Reply:
There are some pushState issues on iOS (https://github.com/aurelia/router/issues/574), and they are solved in the aurelia history api as far as I recall. I think that it is safer to use that if at all possible...

Reply:
OK. Is this something that you would like to do or would you like me to run with this?

Reply:
If you decide to run with this it will need to have unit tests.

Reply:
> What exactly do you mean by breaking the page refresh? Is it #46? I was unable to reproduce that bug from the outlined steps.

Yes. That's the page refresh situation to which I was referring.

Reply:
> Can you reproduce #46?

Yes.

Reply:
> or maybe something like history.navigate(redirectRoute, {trigger: false, replace: true}), see (https://github.com/aurelia/history-browser/blob/master/src/index.js#L126)

Unfortunately, that causing the same error that we're trying to address in the first place, which is the `ERROR [app-router] TypeError: "name is undefined"` problem that this PR is to address.

Reply:
I have gone ahead with the following approach for now: 
```
  // See https://github.com/aurelia-contrib/aurelia-open-id-connect/issues/47
  private redirectTo(instruction: NavigationInstruction, route: string) {
    this.$window.history.pushState({}, '', route);
    instruction.config.redirect = this.openIdConnectConfiguration.logoutRedirectRoute;
  }
```
If you find a better approach to use I would very much like to see it. Thank you for the help on this.

Reply:
Looks good to me...

Reply:
I added your name to the list of contributors.